### ISSUE TITLE
false positive for rows passed to function in a different package

### ISSUE THREAD

**Issue by stevendanna** [jingyugao/rowserrcheck#20]
The following code produces a false positive

```go
> nl -ba repro.go
     1	package repro
     2	
     3	import (
     4		"database/sql"
     5	
     6		"github.com/jingyugao/rowserrcheck/b"
     7	)
     8	
     9	var db *sql.DB
    10	
    11	func issueN() error {
    12		rows, err := db.Query("select 1")
    13		if err != nil {
    14			return err
    15		}
    16		defer rows.Close()
    17		err = b.CheckRows(rows)
    18		if err != nil {
    19			return err
    20		}
    21		return nil
    22	}
 > nl -ba b/util.go
     1	package b
     2	
     3	import "database/sql"
     4	
     5	func CheckRows(rows *sql.Rows) error {
     6		for rows.Next() {
     7	
     8		}
     9		if err := rows.Err(); err != nil {
    10			return err
    11		}
    12		return nil
    13	}
```

produces a false positive:

```
> go vet -vettool ~/rowserrcheck/rowserrcheck ./repro.go
# command-line-arguments
./repro.go:12:23: rows.Err must be checked
```

If CheckRows is moved into the same packages, the problem goes away.

This isn't fixed by #17

**Comment by jingyugao** [comment]
This is because b.CheckRows is external function, so it's block is nil. see https://github.com/golang/tools/blob/master/go/ssa/ssa.go#L259 for more information.

**Comment by stevendanna** [comment]
@jingyugao Yup, thanks for taking a look! I don't think this one is that easy to fix because the analysis framework seems to make whole-program analysis a bit harder unless I've missed some API somewhere. I have a branch that recursively builds the ssa for other packages when we hit them, but I'd worry that it would make execution too slow for the common cases.

**Comment by jingyugao** [comment]
> worry

I think this is govet's problem.
